### ISSUE TITLE
Parse Server Update Issue - MongoDB Connection

### ISSUE THREAD

**Issue by rbwater** [ParsePlatform/parse-server#539]
Hi,

I had successfully set up MongoDB and the Parse Server a few days after the service termination announcement & ultimately had one of my apps successfully running in test mode (i.e. just my development version).  I noticed yesterday that a new version of Parse Server is available with Push Notification API included which I need for another app. Hence I used npm update & npm update -g to update my set up.  Now when I start MongoDB in exactly the same way I did prior to the update & then start the Parse Server with npm start, the server does not make a connection to the MongoDB database that it is pointed at.  If I run mongo to make a connection to the same db_uri, it works and everything is exactly the same as it was pre-update.  My mongod.conf file is exactly as it was pre-update also.  The version of MongoDB seems to be the same as pre-update.

Anyone have any ideas what may have happened & where I should look for a change that is causing this? 

(I suspect something happened to MongoDB because when I tried to migrate data from parse.com for the 2nd app that uses Push Notifications it kept telling me it  could not find the server even though I was using exactly same procedure that succeeded for 1st app).

Also are there any log files for Parse Server? Can't seem to find any.

Thx for any help.

RB

**Comment by drew-gross** [comment]
We reverted a change that url encodes mongo passwords. If you are using certain characters in your mongo password (usually @) you will need to either change your password or wait until we can fix the bug. Please re-open if this is not your issue.

You can turning on some logging with environment variable VERBOSE=1 or by using a logging adapter.

**Comment by rbwater** [comment]
Thanks for your response.  I tried two things:

a) Changing the MongoDB password to a simple combination of lower case letters and numbers & restarting MongoDB with mongo --conf .../mongod.conf & Parse Server with npm start.  Unfortunately, no difference - the server starts fine but does not make a connection to MongoDB.

b) Starting MongoDB without enabling authorization & parse server without username and password in the database_uri.  Unfortunately, this made no difference either.

It seems to be some other issue.  Any other thoughts?  Thx.

RB

**Comment by drew-gross** [comment]
Did you capture any logging output from `VERBOSE=1`?

**Comment by rbwater** [comment]
I did use VERBOSE=1 but could not see anything specific on the Terminal or in the Console.  Is there some other location where I should look for a file?

**Comment by drew-gross** [comment]
It would be in the terminal.

**Comment by rbwater** [comment]
Thanks for assisting with this issue which it turns out may have been a non-issue.  Apologies for any time wasted.

After numerous attempts to get the connection between MongoDB & the update Parse Server to work, I uninstalled both & started over with fresh installations of MongoDB 2.6 (apparently no version 3.0 available for Homebrew) and he latest Parse Server (2.1.11 I believe).  Once everything was set up I started MongoDB in a terminal window & parse server in a second window.  Still no indication from MongoDB that a connection was established.  However, for some reason I decided to send a test request using curl & bingo I got back the info requested & the MongoDB terminal echoed the message saying a connection had been established.  With the earlier version of parse server, the message had appeared when the parse server was started.  After updating, the lack of a message had fooled me into thinking there was no connection so I never tried the curl request! Lesson learned.
